src/fromager/external_commands.py

```diff
@@ -46,6 +47,24 @@ class NetworkIsolationError(subprocess.CalledProcessError):
     pass
 
 
+# GitLab BOT_PAT
+# SECRET, PASSWORD, PASSPHRASE, CRED(entials), TOKEN
+SENSITIVE_KEYS = re.compile(
+    "^(BOT_PAT|.*SECRET.*|.*PASSWORD.*|.*PASSPHRASE.*|.*CRED.*|.*TOKEN.*)$",
```

Comment:
BOT_PAT is pretty specific to our downstream. Does this pattern need to be configurable for other similar cases?

Reply:
Maybe... the `external_commands.run` function does not have access to the `WorkContext`. We either need to refactor a lot of code or come up with a different approach.